### PR TITLE
feat: New setting to specify SameSite cookie mode

### PR DESCRIPTION
**Pull request by mattfawcett**
## Proposed changes

Recent changes to Chrome require setting of SameSite cookie policy if it is acceptable for cookies to be used in a third party setting: https://blog.chromium.org/2020/02/samesite-cookie-changes-in-february.html

Some discussion on this in the community board https://community.ory.sh/t/does-hydra-support-samesite-none-for-cookies/1491

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
- [x] I have read the [security policy](../security/policy)
- [x] I confirm that this pull request does not address a security vulnerability. If this pull request addresses a security
vulnerability, I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation within the code base (if appropriate)
- [x] I have documented my changes in the [developer guide](https://github.com/ory/docs) (if appropriate)

### CODE REVIEW

**Comment by mattfawcett** [comment]
Looks like tests currently fail because `http.SameSiteNoneMode` is only in the latest version of go. Are there any plans to upgrade? Any other advice appreciated - I don't really have much experience with the language.

**Comment by aeneasr** [comment]
Yeah looks like e2e tests are failing but not regular tests. I'll update the image!

**Comment by mattfawcett** [comment]
Sure, will do - thanks

File: driver/configuration/provider_viper.go [hunk 1]
```diff
@@ -48,6 +49,7 @@ const (
 	ViperKeyPublicListenOnHost             = "serve.public.host"
 	ViperKeyPublicListenOnPort             = "serve.public.port"
 	ViperKeyPublicDisableHealthAccessLog   = "serve.public.access_log.disable_for_health"
+	ViperKeyCookieSameSiteMode             = "serve.cookie_same_site_mode"
```
Comment (aeneasr) [inline driver/configuration/provider_viper.go]: Let's actually move this one key down: `serve.cookies.same_site_mode` :)

```suggestion
	ViperKeyCookieSameSiteMode             = "serve.cookies.same_site_mode"
```

**Comment by aeneasr** [comment]
Image is updated - tests pass now :)

**Comment by aeneasr** [comment]
This section in the docs ( https://github.com/ory/docs/pull/273/files#diff-10683e164b12a30b1f69a43baadd57c9R228-R230 ) is actually coming from this file: https://github.com/ory/hydra/blob/master/docs/config.yaml

If you update the config file here with the keys you added there this is all good to merge! :)

**Comment by aeneasr** [comment]
Thank you!

**Pull request merged**